sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreSuite.scala

```diff
@@ -1122,6 +1157,14 @@ abstract class StateStoreSuiteBase[ProviderClass <: StateStoreProvider]
     }
   }
 
+  protected def tryWithProviderResource[T](provider: ProviderClass)(f: ProviderClass => T): T = {
```

Comment:
qq: can we use `Utils.tryWithResource` instead?

Reply:
Yeah that seems to be same. Thanks for the pointer!

Reply:
Hmm, StateStoreProvider has a `close()` method, but not a `Closeable`. Shall we retain this as the change is originally "test only"?